.github/workflows/docs.yml

```diff
@@ -0,0 +1,40 @@
+name: Continuous Integration - Docs
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "**.rs"
+      - "Cargo.toml"
+      - "Cargo.lock"
+  workflow_dispatch:
+
+jobs:
+  docs:
+    name: Generate crate documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+
+      - name: Generate documentation
+        uses: actions-rs/cargo@v1
+        env:
+          RUSTDOCFLAGS: "--enable-index-page -Zunstable-options"
```

Comment:
I suppose we could technically use the `RUSTC_BOOTSTRAP` trick here too if we want to. but I don't have strong opinions

Reply:
I'm afraid a straight-up reimplementation didn't cut it:
https://github.com/huitseeker/narwhal/runs/6370782912?check_suite_focus=true